gurubu-client/src/app/components/common/button/index.tsx

```diff
@@ -0,0 +1,97 @@
+import classNames from "classnames";
+import React, {
+  forwardRef,
+  ButtonHTMLAttributes,
+  AnchorHTMLAttributes,
+} from "react";
+
+type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;
+type AnchorProps = AnchorHTMLAttributes<HTMLAnchorElement>;
+
+type CommonProps = {
+  children: React.ReactNode;
+  variant?: "primary" | "secondary";
+  className?: string;
+};
+
+type CustomButtonProps = CommonProps &
+  (
+    | ({
+        as: "button";
+        onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
+      } & ButtonProps)
```

Comment:
```suggestion
      } & ButtonProps & { href: string })
```

Reply:
Hi! @mertcanaltin! I see that with this typing, button component is available to use both href and onClick props without getting any ts error. And also the button component uses that href attribute and I can see that attribute in the DOM. This may be cause accessibility conflicts with screen readers. What is your suggestion? 
<img width="694" alt="image" src="https://github.com/Trendyol/gurubu/assets/65610514/1b99243b-e65b-41dd-bc3e-86a6b3e3f1e2">

Reply:
greetings, what I was talking about was actually to lighten the condition in the #86 (comment) field, it would be healthy to take this step to prevent the warning on the type side, as in line 23